src/ImageSharp.Web/Commands/PresetOnlyQueryCollectionRequestParser.cs

```diff
@@ -0,0 +1,59 @@
+﻿// Copyright (c) Six Labors.
+// Licensed under the Apache License, Version 2.0.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.AspNetCore.Http;
+using Microsoft.AspNetCore.WebUtilities;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Primitives;
+
+namespace SixLabors.ImageSharp.Web.Commands
+{
+    /// <summary>
+    /// Parses preset name from the request querystring and returns the commands configured for that preset.
+    /// </summary>
+    public class PresetOnlyQueryCollectionRequestParser : IRequestParser
+    {
+        private readonly IDictionary<string, IDictionary<string, string>> presets;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="PresetOnlyQueryCollectionRequestParser"/> class.
+        /// </summary>
+        /// <param name="presetOptions">The preset options.</param>
+        public PresetOnlyQueryCollectionRequestParser(IOptions<PresetOnlyQueryCollectionRequestParserOptions> presetOptions) =>
+            this.presets = ParsePresets(presetOptions.Value.Presets);
+
+        /// <inheritdoc/>
+        public IDictionary<string, string> ParseRequestCommands(HttpContext context)
+        {
+            if (context.Request.Query.Count == 0)
+            {
+                return new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
+            }
+
+            var requestedPreset = context.Request.Query["preset"][0];
```

Comment:
If you write a test for parsing commands that do not contain "preset" you'll see that this throws `IndexOutOfrangeException`. I'd rewrite is as the following.

```c#
public IDictionary<string, string> ParseRequestCommands(HttpContext context)
{
    // I'd probably make this a public constant under the name QueryKey
	const string queryKey= "preset";
	IQueryCollection query = context.Request.Query;
	if (query.Count == 0 || !query.ContainsKey(queryKey))
	{
		return new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
	}

	string requestedPreset = context.Request.Query[queryKey][0];
	return this.presets.GetValueOrDefault(requestedPreset) ?? new Dictionary<string, string>(StringComparer.OrdinalIgnoreCase);
}

Reply:
We probably also want a test for this? And maybe also another one for case insensitivity?

Reply:
I actually did have a test for that, but forgot to run it (and the GitHub pipeline doesn't run automatically for me, because I'm a first-time contributor).

It's fixed now.

Reply:
I also added the case insensitivity test (which tests case insensitivity for both the query param name, and the preset name).

Reply:
Thanks for adding these @marklagendijk.